File: Algorithms/python/48 - Rotate Image.py
class Solution(object):
    def rotate(self, matrix):
        """
        :type matrix: List[List[int]]
        :rtype: void Do not return anything, modify matrix in-place instead.
        """
        
        n = len(matrix)
        temp = [row[:] for row in matrix]
        
        for i in range(0, n):
            for j in range(0, n):
                matrix[j][n - 1 - i] = temp[i][j]

File: Algorithms/cpp/48 - Rotate Image.cpp
class Solution {
public:
    void rotate(vector<vector<int>>& matrix) {
        // var
        int n = matrix.size();
        
        // mirro
        for (int i = 0; i < n; i++)
        for (int j = 0; j < n; j++)
            matrix[i].push_back(matrix[n - 1 - j][n - 1 - i]);
        
        // remove old
        for (int i = 0; i < n; i++)
            matrix[i].assign(matrix[i].begin() + n, matrix[i].end());
        
        // 180
        for (int i = 0; i < n / 2; i++)
            swap(matrix[i], matrix[n - 1 - i]);
    }
};